contracts/sg-ics721/src/execute.rs

```diff
@@ -16,28 +20,52 @@ impl Ics721Execute for SgIcs721Contract {
             contract_info,
             name,
             symbol,
+            extension: _, // ignore extension coming from standard cw721, since sg721 has its own extension (collection info)
             num_tokens,
         } = get_collection_data(deps, sender)?;
         let collection_info: CollectionInfoResponse = deps
             .querier
             .query_wasm_smart(sender, &QueryMsg::CollectionInfo {})?;
+        let royalty_info = collection_info.royalty_info.map(|r| RoyaltyInfo {
+            payment_address: Addr::unchecked(r.payment_address),
+            share: r.share,
+        });
+        let extension = Some(CollectionExtension {
+            description: collection_info.description,
+            image: collection_info.image,
+            external_link: collection_info.external_link,
+            explicit_content: collection_info.explicit_content,
+            start_trading_time: collection_info.start_trading_time,
+            royalty_info,
+        });
```

Comment:
Here sg-ics721 overrides default implementation and gets collection data from sg721-base via `CollectionInfo` query.